controllers/topology/machinedeployment_controller.go

```diff
@@ -148,5 +149,29 @@ func (r *MachineDeploymentReconciler) reconcileDelete(ctx context.Context, md *c
 		return ctrl.Result{}, errors.Wrapf(err, "failed to patch %s", tlog.KObj{Obj: md})
 	}
 
+	// If the MachineDeployment has a MachineHealthCheck delete it.
+	if err := r.deleteMachineHealthCheckForMachineDeployment(ctx, md); err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
Let's do this operation before removing the finalizers, otherwise there can be orphan MHC left around